speculos/client.py

```diff
@@ -33,14 +33,32 @@ def check_status_code(response: requests.Response, url: str) -> None:
 
 class ApduResponse:
     def __init__(self, response: requests.Response) -> None:
-        self.response = response
-
-    def receive(self) -> bytes:
-        check_status_code(self.response, "/apdu")
-        data, status = split_apdu(bytes.fromhex(self.response.json()["data"]))
-        if status != 0x9000:
-            raise ApduException(status, data)
-        return data
+        self.__response = response
+
+    def __receive(self):
+        """ Get the response if not yet received and store it in `__data` and `__status` attributes """
+        if self.__response is not None:
+            check_status_code(self.__response, "/apdu")
+            self.__data, self.__status = split_apdu(bytes.fromhex(self.__response.json()["data"]))
```

Comment:
[Code style] I personally prefer when the function calls are not to much nested. For example I prefer code such as:
```python
json_resp = self.__response.json()
apdu_message = bytes.fromhex(json_resp["data"]))
self.__data, self.__status = split_apdu(apdu_message)
```
... because by doing so, when one of the call fails and raises an exception, the exception call trace indicates precisely which function failed.
Anyway this is my personal opinion, and feel free to follow such advice or keep your code as-is.

Reply:
We can integrate that change in the PR as well.